DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php

```diff
@@ -279,7 +279,7 @@ public function applyDefaults(ContainerBuilder $container, $serviceId, array $at
             'menu_factory' => 'knp_menu.factory',
             'route_builder' => 'sonata.admin.route.path_info'.
                 (($manager_type == 'doctrine_phpcr') ? '_slashes' : ''),
-            'label_translator_strategy' => 'sonata.admin.label.strategy.native',
+            'label_translator_strategy' => null !== $settings['label_translator_strategy'] ? $settings['label_translator_strategy'] : 'sonata.admin.label.strategy.native',
```

Comment:
Isn't [this code](https://github.com/sonata-project/SonataAdminBundle/blob/3.x/DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php#L296) supposed to achieve the exact same goal?

Reply:
you mean **$definition->addMethodCall($method, $args);** ?
currently **'label_translator_strategy' => 'sonata.admin.label.strategy.native'** is hardcoded, and there is no way to globaly change the stratagy, only one by one for each admin service

Reply:
My bad, I missed the fact that `$settings` is not `$overwriteAdminConfiguration`.

Anyway, if other properties `sonata_admin.admin_services.*` work out of the box, I don't get why you have to override this one here.

Reply:
not sure about other properties yet. I was needed to change the label translation strategy for all my admin services and faced this issue

Reply:
but i guess this [settings](https://github.com/sonata-project/SonataAdminBundle/blob/3.x/DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php#L274-L279) will have similar issue if will try to set them in a global configuration area

Reply:
That's why I asked. Could you test if they work? If not we will need a fix for all of them with a cleaner solution.